File: trafficsvr/src/main/java/com/example/back/recommended/recommendedService/RecommendedService.java
package com.example.back.recommended.recommendedService;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import com.example.back.recommended.recommended.Recommended;
import com.example.back.recommended.recommendedRepository.RecommendedRepository;

@Service
public class RecommendedService {
	
	private final RecommendedRepository recommendedRepository;

	@Autowired
	public RecommendedService(RecommendedRepository recommendedRepository) {
		super();
		this.recommendedRepository = recommendedRepository;
	}

	public Recommended addRecommended(Recommended recommended) { return recommendedRepository.save(recommended); }
	public void clearAllRecommended() { recommendedRepository.deleteAll(); }
	public void deleteRecommendation(Recommended recommended) { recommendedRepository.delete(recommended); }
	public void deleteById(Long id) { recommendedRepository.deleteById(id); }
	public List<Recommended> getAllRecommended() { return recommendedRepository.findAll(); }
	public List<Recommended> getJunctionInstances(String junction) { return recommendedRepository.getJunctionInstances(junction); }
	public List<Recommended> getJunctionWithStartYear(String junction, Integer startYear) {
		return recommendedRepository.getJunctionWithStartYear(junction, startYear);
	}
	public List<Recommended> getDistrictInstancesWithStartYear(String district, Integer startYear) {
		return recommendedRepository.getDistrictInstancesWithStartYear(district, startYear);
	}

}

File: TRAFFIC_PREDICTION_CLI/src/app/services/db/recommended/recommended.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Injectable({
	providedIn: 'root'
})
export class RecommendedService {
	constructor(private http: HttpClient) { }

	public checkIfAlreadyRecommended(junction: string, startYear: number) {
		return this.http.get('recommended/checkIfAlreadyRecommended?junction=' + junction + '&startYear=' + startYear)
	}

	public storeRecommendation(recommended: Object) {
		return this.http.post('recommended/addRecommended', recommended)
	}

	public clearAllRecommended() {
		return this.http.get('recommended/clearAllRecommended')
	}

	public deleteRecommendation(junction: string, district: string, startYear: number, durationYears: number) {
		return this.http.get('recommended/deleteRecommendation?junction=' + junction
			+ '&district=' + district 
			+ '&startYear=' + startYear 
			+ '&durationYears=' + durationYears
		)
	}

	public getDistrictInstancesWithStartYear(district: string, startYear: number) {
		return this.http.get('recommended/getDistrictInstancesWithStartYear?district=' + district + '&startYear=' + startYear)
	}
	
}
